tests/collections/collection.test.ts

```diff
@@ -383,6 +429,95 @@ for (const testClient in testClients) {
+        docList.forEach((doc, index) => {
+          doc.username = doc.username+(index+1);
+        });
+        const res = await collection.insertMany(docList);    
+        assert.strictEqual(res.insertedCount, docList.length);
+        assert.strictEqual(res.acknowledged, true);
+        assert.strictEqual(_.keys(res.insertedIds).length, 20);
+        
+        //const idToUpdateAndCheck = sampleDocsWithIdList[0]._id;
+        const updateManyResp = await collection.updateMany({"city": "la"},
+                              {
+                                "$set" : { "state" : "ca" }
+                              });     
+        assert.strictEqual(updateManyResp.matchedCount, 0);
+        assert.strictEqual(updateManyResp.modifiedCount, 0);
+        assert.strictEqual(updateManyResp.acknowledged, true);
+        assert.strictEqual(updateManyResp.upsertedCount, undefined);
+        assert.strictEqual(updateManyResp.upsertedId, undefined);
+      });
+      //TODO skipped until https://github.com/stargate/jsonapi/issues/273 is fixed
+      it.skip('should upsert with upsert flag set to true when not found', async () => {
+        let docList = Array.from({ length: 2 }, ()=>({username: "id", city : "nyc"}));
+        docList.forEach((doc, index) => {
+          doc.username = doc.username+(index+1);
+        });
+        const res = await collection.insertMany(docList);    
+        assert.strictEqual(res.insertedCount, docList.length);
+        assert.strictEqual(res.acknowledged, true);
+        assert.strictEqual(_.keys(res.insertedIds).length, 2);
+        
+        //const idToUpdateAndCheck = sampleDocsWithIdList[0]._id;
+        const updateManyResp = await collection.updateMany({"city": "la"},
+                              {
+                                "$set" : { "state" : "ca" }
+                              },
+                              {
+                                "upsert": true
+                              });     
+        assert.strictEqual(updateManyResp.matchedCount, 0);
+        assert.strictEqual(updateManyResp.modifiedCount, 0);
+        assert.strictEqual(updateManyResp.acknowledged, true);
+        assert.strictEqual(updateManyResp.upsertedCount, 1);
+        assert.ok(updateManyResp.upsertedId);
+      });
+      it('should fail when moreData returned by updateMany as true', async () => {
+        let docList = Array.from({ length: 25 }, ()=>({username: "id", city : "nyc"}));
+        docList.forEach((doc, index) => {
+          doc.username = doc.username+(index+1);
+        });
+        const res = await collection.insertMany(docList);    
+        assert.strictEqual(res.insertedCount, docList.length);
+        assert.strictEqual(res.acknowledged, true);
+        assert.strictEqual(_.keys(res.insertedIds).length, 25);
+        
+        //const idToUpdateAndCheck = sampleDocsWithIdList[0]._id;
+        try{
+          const updateManyResp = await collection.updateMany({"city": "nyc"},
```

Comment:
Question: are the updates "rolled back" if `moreData` is returned? Or does stargate update 20 documents, and then return `moreData` to tell the client that it couldn't update everything in one shot?

Reply:
It's the later. Stargate updates 20 documents and then returns `moreData` as `true` if it finds more records to be updated.

Reply:
Thanks for clarifying.